src/app/components/guards/auth.tsx

```diff
@@ -1,22 +1,48 @@
-import useHasStateRehydrated from '@hooks/stores/useHasRehydrated';
+import { ReactNode, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import useSecretKey from '@hooks/useSecretKey';
 import useWalletSelector from '@hooks/useWalletSelector';
-import { Navigate } from 'react-router-dom';
 
-function AuthGuard({ children }: React.PropsWithChildren) {
-  const { encryptedSeed, seedPhrase } = useWalletSelector();
-  const hydrated = useHasStateRehydrated();
+interface AuthGuardProps {
+  children: ReactNode;
+}
 
-  if (hydrated && encryptedSeed && !seedPhrase) {
-    return <Navigate to="/login" />;
-  }
+function AuthGuard({ children }: AuthGuardProps) {
+  const navigate = useNavigate();
+  const { masterPubKey, encryptedSeed } = useWalletSelector();
+  const { getSeed, hasSeed } = useSecretKey();
+  const [authTested, setAuthTested] = useState(false);
 
-  if (hydrated && !encryptedSeed) {
-    return <Navigate to="/landing" />;
-  }
+  const restoreSession = async () => {
+    if (encryptedSeed) {
+      navigate('/login');
+      setAuthTested(true);
+      return;
+    }
+    const hasSeedPhrase = await hasSeed();
+    if (!hasSeedPhrase || !masterPubKey) {
+      navigate('/landing');
+      setAuthTested(true);
```

Comment:
I don't think this does anything. Once you navigate, the local state is lost.

Reply:
resolved however i found if we don't return after navigating the effect continues to check for the remaining conditions and ends up defaulting to the login screen

Reply:
That makes sense. I think returning is the better approach.